xds/internal/xdsclient/client_refcounted.go

```diff
@@ -80,22 +71,29 @@ func newRefCountedWithConfig(fallbackConfig *bootstrap.Config) (XDSClient, func(
 		config = fallbackConfig
 	} else {
 		var err error
-		config, err = bootstrapNewConfig()
+		config, err = bootstrap.NewConfig()
 		if err != nil {
 			return nil, nil, fmt.Errorf("xds: failed to read bootstrap file: %v", err)
 		}
 	}
 
+	if c := clients[name]; c != nil {
+		c.incrRef()
+		return c, grpcsync.OnceFunc(func() { clientRefCountedClose(name) }), nil
+	}
+
 	// Create the new client implementation.
-	c, err := newWithConfig(config, defaultWatchExpiryTimeout, defaultIdleAuthorityDeleteTimeout)
+	c, err := newClientImpl(config, watchExpiryTimeout, idleAuthorityTimeout)
 	if err != nil {
 		return nil, nil, err
 	}
-	singletonClient = &clientRefCounted{clientImpl: c, refCount: 1}
-	singletonClientImplCreateHook()
+	c.logger.Infof("Created client with name %q to xDS management server: %q", name, config.XDSServers()[0])
```

Comment:
I'm having a hard time understanding why it's always at index 0 with respect to this dimension here: "Current gRPC xDS implementations only use the first xDS configuration server listed in the bootstrap JSON document. Fallback implementation requires changes to use all servers listed in order from highest to lowest priority. Priority is assigned based on server order in bootstrap.json." Is this handled at a higher layer, and once called into this it is expected to only have one xDS Server in the config?

Reply:
Right, I meant what you logged. My point was that once you actually implement fallback, does reading at index 0 still make sense for logging? Or will you change this once that change happens, or is that irrelevant because the caller of this handles fallback?

Reply:
I don't think it really matters that we include all the xds server names in this log message. But I changed the log message to indicate that the server being printed is the primary.